### PR TITLE
add craft middleware command

### LINKED ISSUE #293: craft middleware MiddlewareName : Create a new middleware

### PR DESCRIPTION
**Pull request by vaibhavmule**
Closes #293

### CODE REVIEW

File: masonite/commands/MiddlewareCommand.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import os
+
+from masonite.view import View
+from masonite.app import App
+from masonite.helpers.filesystem import make_directory
+
+from cleo import Command
+
+
+class MiddlewareCommand(Command):
```
Comment (josephmancuso) [inline masonite/commands/MiddlewareCommand.py]: This should look more like this file https://github.com/MasoniteFramework/core/blob/develop/masonite/commands/ModelCommand.py

File: masonite/snippets/scaffold/middleware.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+""" {{ class }} Middleware """
+
+class {{ class }}Middleware:
+    """{{ class }} Middleware
+    """
+
+    def __init__(self, Request):
```
Comment (josephmancuso) [inline masonite/snippets/scaffold/middleware.html]: 2.1 doesn't support Parameter Resolving anymore. You'll need to do:

```python
from masonite.request import Request
...
def __init__(self, request: Request):
```

File: masonite/snippets/scaffold/middleware.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        """ Inject Any Dependencies From The Service Container """
```
Comment (josephmancuso) [inline masonite/snippets/scaffold/middleware.html]: Again follow the docstring here like in the core repo. Example here https://github.com/MasoniteFramework/masonite/blob/develop/app/http/middleware/AuthenticationMiddleware.py#L10-L15

File: masonite/snippets/scaffold/middleware.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        self.request = Request
+
+    def before(self):
+        """ Run This Middleware Before The Route Executes """
+        pass
+
+    def after(self):
+        """" Run This Middleware After The Route Executes """
```
Comment (josephmancuso) [inline masonite/snippets/scaffold/middleware.html]: docstring is messed up. Has too many `""""` Also let's follow the docstrings I've been using in other files:

```python
    def after(self):
        """Run This Middleware After The Route Executes
        """
```

File: masonite/commands/MiddlewareCommand.py (hunk 1 above)
Comment (josephmancuso) [reply masonite/commands/MiddlewareCommand.py]: 🤔 no lol. Maybe we should add that way in this pr?

Just need to take a look at the BaseScaffoldCommand and maybe add something like:

```
...
postfix='Middleware'
```

and then we can add that to scaffolding commands to postfix stuff.

Comment (josephmancuso) [reply masonite/commands/MiddlewareCommand.py]: you'll need to add the logic to append file names in that `BaseScaffoldCommand`.

Comment (vaibhavmule) [reply masonite/commands/MiddlewareCommand.py]: How about `suffix` true.

```
...
suffix = True
```

File: masonite/snippets/scaffold/middleware.html (hunk 4 above)
Comment (vaibhavmule) [reply masonite/snippets/scaffold/middleware.html]: @josephmancuso Made the changes as requested.

**Comment by vaibhavmule** [comment]
Oh pep8, let me fix it that.

**Pull request merged**